cloud/scope/cluster.go

```diff
@@ -143,15 +144,42 @@ func (s *ClusterScope) DeleteVPC() error {
 }
 
 func (s *ClusterScope) ensureVPCUnique(vpcName string) (*vpcv1.VPC, error) {
-	listVpcsOptions := &vpcv1.ListVpcsOptions{}
-	vpcs, _, err := s.IBMVPCClient.ListVpcs(listVpcsOptions)
-	if err != nil {
+	var vpc *vpcv1.VPC
+	f := func(start string) (bool, string, error) {
+		// check for existing vpcs
+		listVpcsOptions := &vpcv1.ListVpcsOptions{}
+		if start != "" {
+			listVpcsOptions.Start = &start
+		}
+
+		vpcsList, _, err := s.IBMVPCClient.ListVpcs(listVpcsOptions)
+		if err != nil {
+			return false, "", err
+		}
+
+		if vpcsList == nil {
+			return false, "", fmt.Errorf("vpc list returned is nil")
+		}
+
+		for i, v := range vpcsList.Vpcs {
+			if (*v.Name) == vpcName {
+				vpc = &vpcsList.Vpcs[i]
+				return true, "", nil
+			}
+		}
+
+		if vpcsList.Next != nil && *vpcsList.Next.Href != "" {
+			return false, *vpcsList.Next.Href, nil
+		}
+		return true, "", nil
+	}
+
+	if err := utils.PagingHelper(f); err != nil {
 		return nil, err
 	}
-	for _, vpc := range vpcs.Vpcs {
-		if (*vpc.Name) == vpcName {
-			return &vpc, nil
-		}
+
+	if vpc != nil {
```

Comment:
IMO you can directly return vpc, since anyway you are validating for nil in called func.

Reply:
It seems, the same can be applied to all ensure func.

Reply:
Fixed.